### PR TITLE
counting packages sent

### PR DESCRIPTION
**Pull request by adenhaus**
## Description
Counting json packages sent from client. Intention is to count packages received on base station side as well and make sure they match.

### CODE REVIEW

**Comment by MiltFra** [comment]
Do we want to have this on master?

**Comment by MiltFra** [comment]
So I've just had a think about this: This is actually a good number to have and to send to the base station. We can combine it with the timestamp into a primary key if we ever want to have a database of all runs and it allows us to add a timestamp the moment we receive it to analyse delays. Timestamps are not guaranteed to be unique, especially if we're dealing in micros or even milis.

File: src/telemetry/sendloop.cpp [hunk 1]
```diff
@@ -28,15 +29,20 @@ void SendLoop::run()
...
       log_.ERR("Telemetry", "Error sending message");
-      data::Telemetry telemetry_data = data_.getTelemetryData();
       telemetry_data.module_status   = data::ModuleStatus::kCriticalFailure;
       data_.setTelemetryData(telemetry_data);
 
       break;
     }
 
+    // Increment num packages sent
+    telemetry_data.num_packages_sent += 1;
```
Comment (MiltFra) [inline src/telemetry/sendloop.cpp:43]: I'm not sure we should write this to data. This is not useful information to anyone but the writer. Writes to data are much more expensive than local variables (or class members).

**Review by MiltFra — CHANGES_REQUESTED** [review]

File: src/telemetry/sendloop.cpp [hunk 2]
```diff
@@ -28,15 +29,20 @@ void SendLoop::run()
...
     if (!main_ref_.client_.sendData(writer.getString())) {
       log_.ERR("Telemetry", "Error sending message");
-      data::Telemetry telemetry_data = data_.getTelemetryData();
       telemetry_data.module_status   = data::ModuleStatus::kCriticalFailure;
       data_.setTelemetryData(telemetry_data);
 
       break;
     }
 
+    // Increment num packages sent
```
Comment (MiltFra) [inline src/telemetry/sendloop.cpp:42]: This reminds me of the textbook example of pointless comments. 

[E.g.](https://stackoverflow.blog/2021/07/05/best-practices-for-writing-code-comments/) this is called the "canonical bad example"
```
i = i + 1;         // Add one to i
```

Comment (adenhaus) [reply src/telemetry/sendloop.cpp:42]: Lol fair enough

File: src/telemetry/sendloop.cpp (hunk 1 above)
Comment (adenhaus) [reply src/telemetry/sendloop.cpp:43]: Okay understood, I'll just pack it in with the json and send it

**Comment by adenhaus** [comment]
@MiltFra what's the issue with the failing build? Local tests pass.

**Review by MiltFra — CHANGES_REQUESTED** [review]
Please fix your auto-formatting; you're changing lines without any reason.

**Comment by adenhaus** [comment]
> Please fix your auto-formatting; you're changing lines without any reason.

File: src/telemetry/writer.hpp [hunk 3]
```diff
@@ -31,6 +31,7 @@ class Writer {
 
   // separate functions that allow to better manage data points based on their purpose
   void packTime();
+  void packID(uint16_t id);
```
Comment (MiltFra) [inline src/telemetry/writer.hpp:34]: should be `packId`

File: src/telemetry/writer.cpp [hunk 4]
```diff
@@ -18,6 +18,12 @@ void Writer::packTime()
                         .count());
 }
 
+void Writer::packID(uint16_t id)
```
Comment (MiltFra) [inline src/telemetry/writer.cpp:21]: `const`

File: src/telemetry/sendloop.cpp [hunk 5]
```diff
@@ -18,25 +20,31 @@ void SendLoop::run()
...
-      data::Telemetry telemetry_data = data_.getTelemetryData();
-      telemetry_data.module_status   = data::ModuleStatus::kCriticalFailure;
+      telemetry_data.module_status = data::ModuleStatus::kCriticalFailure;
       data_.setTelemetryData(telemetry_data);
 
       break;
     }
 
+    num_packs_sent += 1;
+
```
Comment (MiltFra) [inline src/telemetry/sendloop.cpp:47]: there's lots of whitespaces around; wondering whether that makes sense? I know you're just fitting your changes into the style that was already there so I suppose it's fine.

File: src/telemetry/sendloop.cpp [hunk 6]
```diff
@@ -18,25 +20,31 @@ void SendLoop::run()
 {
   log_.DBG("Telemetry", "Telemetry SendLoop thread started");
 
+  uint16_t num_packs_sent = 0;
```
Comment (MiltFra) [inline src/telemetry/sendloop.cpp:23]: I'd call it `num_packages_sent` but idk, won't stop me from approving so dw

**Review by MiltFra — CHANGES_REQUESTED** [review]

**Review by MiltFra — APPROVED** [review]

**Review by sashank17 — APPROVED** [review]

**Comment by adenhaus** [comment]
Hi @MiltFra please merge thanks boss

**Pull request merged**